crates/ruff_python_parser/tests/snapshots/invalid_syntax@for_stmt_missing_target.py.snap

```diff
@@ -0,0 +1,60 @@
+---
+source: crates/ruff_python_parser/tests/fixtures.rs
+input_file: crates/ruff_python_parser/resources/inline/err/for_stmt_missing_target.py
+---
+## AST
+
+```
+Module(
+    ModModule {
+        range: 0..14,
+        body: [
+            For(
+                StmtFor {
+                    range: 0..13,
+                    is_async: false,
+                    target: Name(
+                        ExprName {
+                            range: 4..6,
+                            id: "in",
+                            ctx: Store,
+                        },
+                    ),
+                    iter: Name(
```

Comment:
We should avoid calling `parse_identifier` if the current token is `in` for a better recovery.

Reply:
Yeah, although that means we need to fill the `target` with an empty `ExprName`

Reply:
That's true. But I think it improves the error message (and `in` isn't the variable name)

Reply:
I think the underlying problem is a bit different. The expression parsing fallback in case of keyword is to convert it into a Name node which is incorrect. I think the reason for this logic was to have better recovery for `pass = 1` but they can be done locally in `parse_assignment_statement`.